automata/base/utils.py

```diff
@@ -23,6 +23,26 @@ def freeze_value(value):
         return tuple(freeze_value(element) for element in value)
     return value
 
+def get_renaming_function(counter):
+    """
+    A helper function that returns a renaming function to be used in the creation of
+    other automata. The parameter counter should be an itertools count.
+    This helper function will return the same distinct output taken from counter
+    for each distinct input.
+    """
+
+    new_state_name_dict = {}
+
+    def renaming_function(item):
+        nonlocal new_state_name_dict
```

Comment:
@eliotwrobson You're not reassigning `new_state_name_dict` within this function, so the `nonlocal` declaration should be unnecessary.